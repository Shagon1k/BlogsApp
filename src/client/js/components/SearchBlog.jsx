import React from "react";

class SearchBlog extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
            value: '',
            option: 'title'
        };
	}

	handleChange(e) {
		this.setState({value: e.target.value}, function() {
			this.props.onSearchChange(this.state)
		})
	}

	handleOptionChange(e) {
		this.setState({option: e.target.value}, function() {
			this.props.onSearchChange(this.state)
		})
	}

	render() {
		return (
			<div className="searchBlog">
                <input className="searchBlogInput" type="text" placeholder="Search" value={this.state.value} onChange={(e) => this.handleChange(e)} />
            	<form className="searchBlogOption">
            		<label>
            			<input type="radio" value="title" checked={this.state.option === 'title'} onChange={(e) => this.handleOptionChange(e)} />
            			by title
            		</label>
            		<label>
            			<input type="radio" value="author" checked={this.state.option === 'author'} onChange={(e) => this.handleOptionChange(e)} />
            			by author
            		</label>
            	</form>
            </div>
			);
	}
}

export default SearchBlog;