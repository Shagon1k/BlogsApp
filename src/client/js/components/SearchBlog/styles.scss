.searchBlog {
	display: flex;
	flex-flow: row wrap;
	align-items: center;
	justify-content: space-between;
	width: 100%;
	margin: 20px 0;

	.searchBlogInput {
		flex-basis: 70%;
		height: 35px;
		font-size: 20px;
	}

	.searchBlogOption {
		display: flex;
		flex-flow: row wrap;
		align-items: center;
		justify-content: center;
		flex-basis: 25%;
		
		label {
			flex-basis: 100%;
			color: white;
			cursor: pointer;
		}
	}
}