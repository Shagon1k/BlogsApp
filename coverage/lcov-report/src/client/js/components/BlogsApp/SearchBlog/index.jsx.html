<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/BlogsApp/SearchBlog/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/BlogsApp/SearchBlog</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import './styles.scss';
&nbsp;
const SearchBlog = ({search, onValueChange, onParamChange}) =&gt; (
	&lt;div className="searchBlog"&gt;
		&lt;input className="searchBlogInput" type="text" placeholder="Search" value={search.searchValue} onChange={(e) =&gt; onValueChange(e.target.value)} /&gt;
		&lt;form className="searchBlogOption"&gt;
			&lt;label&gt;
				&lt;input type="radio" value="BY_TITLE" checked={search.searchParam === 'BY_TITLE'} onChange={(e) =&gt; onParamChange(e.target.value)} /&gt;
				by title
			&lt;/label&gt;
			&lt;label&gt;
				&lt;input type="radio" value="BY_AUTHOR" checked={search.searchParam === 'BY_AUTHOR'} onChange={(e) =&gt; onParamChange(e.target.value)} /&gt;
				by author
			&lt;/label&gt;
		&lt;/form&gt;
	&lt;/div&gt;
)
&nbsp;
export default SearchBlog;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span></pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
