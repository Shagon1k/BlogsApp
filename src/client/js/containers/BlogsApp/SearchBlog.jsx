import { connect } from 'react-redux';
import { setSearchValue, setSearchParam } from '../../actions';
import SearchBlog from '../../components/BlogsApp/SearchBlog/index.jsx';

const mapStateToProps = state => ({
	search: state.search
})

const mapDispatchToProps = dispatch => {
	return {
		onValueChange: value => {
			dispatch(setSearchValue(value))
		},
		onParamChange: param => {
			dispatch(setSearchParam(param))
		}
	}
}

const SearchBlogContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(SearchBlog)

export default SearchBlogContainer;