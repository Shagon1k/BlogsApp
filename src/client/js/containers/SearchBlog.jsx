import { connect } from 'react-redux';
import { setSearchValue, setSearchParam } from '../actions';
import SearchBlog from '../components/SearchBlog/index.jsx';

const mapStateToProps = state => ({
	search: state.search
})

const mapDispatchToProps = dispatch => {
	return {
		onValueChange: value => {
			dispatch(setSearchValue(value))
		},
		onParamChange: param => {
			dispatch(setSearchParam(param))
		}
	}
}

const SearchBlogContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(SearchBlog)

export default SearchBlogContainer;