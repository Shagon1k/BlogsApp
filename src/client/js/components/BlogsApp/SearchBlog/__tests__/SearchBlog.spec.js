import React from 'react';
import SearchBlog from '../index.jsx';

describe('SearchBlog component', () => {
	const search = {
		searchParam: 'BY_TITLE',
		searchValue: ''
	};
	const onValueChange = jest.fn();
	const onParamChange = jest.fn();

	it('renders correctly', () => {
		const wrapper = shallow(<SearchBlog onValueChange={onValueChange} onParamChange={onParamChange} search={search}/>);
		expect(wrapper).toMatchSnapshot();

	});
	
	it('triggers onParamValue function correctly', () => {
		const wrapper = shallow(<SearchBlog onValueChange={onValueChange} onParamChange={onParamChange} search={search}/>);
		wrapper.find('.searchBlogInput').simulate('change', {target: {value: 'ololo'}});
		expect(onValueChange).toBeCalled();
	})
})