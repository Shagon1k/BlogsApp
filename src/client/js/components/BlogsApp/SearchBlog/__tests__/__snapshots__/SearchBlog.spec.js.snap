// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBlog component renders correctly 1`] = `
<div
  className="searchBlog"
>
  <input
    className="searchBlogInput"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <form
    className="searchBlogOption"
  >
    <label>
      <input
        checked={true}
        onChange={[Function]}
        type="radio"
        value="BY_TITLE"
      />
      by title
    </label>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="BY_AUTHOR"
      />
      by author
    </label>
  </form>
</div>
`;
